src/msg/async/AsyncConnection.cc

```diff
@@ -283,11 +283,12 @@ ssize_t AsyncConnection::read_bulk(char *buf, unsigned len)
 ssize_t AsyncConnection::write(bufferlist &bl,
                                std::function<void(ssize_t)> callback,
                                bool more) {
-
-    std::unique_lock<std::mutex> l(write_lock);
+    send_lock.lock();
```

Comment:
There is no need to protect _try_send and outcoming_bl with write_lock, see `ProtocolV1::write_event` for example.